concrete/dense_directed_matrix.go

```diff
@@ -112,8 +115,16 @@ func (g *DirectedDenseGraph) HasEdgeFromTo(u, v graph.Node) bool {
 	return uid != vid && !isSame(g.mat.At(uid, vid), g.absent)
 }
 
-func (g *DirectedDenseGraph) Weight(e graph.Edge) float64 {
-	return g.mat.At(e.From().ID(), e.To().ID())
+func (g *DirectedDenseGraph) Weight(x, y graph.Node) (w float64, ok bool) {
```

Comment:
The naming differences are significant. u->v or u--v exists, but x->y/x--y may not. At some point when the docs are gone over this should be made clear.